src/components/autocomplete/js/autocompleteController.js

```diff
@@ -327,9 +334,8 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
 
         // cancel results if search text is not long enough
         if (!isMinLengthMet()) {
-          ctrl.loading = false;
           ctrl.matches = [];
-          ctrl.hidden  = shouldHide();
+          setLoading(false);
```

Comment:
The convention for this I've been using is to use `defineProperty` to create a watcher when the value is changed.  See line 23.